morango/utils/sync_utils.py

```diff
@@ -207,11 +227,9 @@ def _queue_into_buffer(transfersession):
         queue_rmc_buffer = """INSERT INTO {outgoing_rmcb}
                             (instance_id, counter, transfer_session_id, model_uuid)
                             SELECT instance_id, counter, '{transfer_session_id}', store_model_id
-                            FROM {record_max_counter} AS rmc, {outgoing_buffer} AS buffer
-                            WHERE EXISTS (SELECT 1
-                                          FROM {outgoing_buffer}
-                                          WHERE buffer.model_uuid = rmc.store_model_id)
-                                          AND buffer.transfer_session_id = '{transfer_session_id}'
+                            FROM {record_max_counter} AS rmc LEFT JOIN {outgoing_buffer} AS buffer
+                            WHERE buffer.model_uuid = rmc.store_model_id
+                            AND buffer.transfer_session_id = '{transfer_session_id}'
```

Comment:
The old query took an incredibly long time. This new query passes tests and is way faster.

Reply:
Cool, can't see any issues! :racehorse: